apps/showcase/src/components/training/code-editor-view/code-editor-view.component.ts

```diff
@@ -195,7 +195,9 @@ export class CodeEditorViewComponent implements OnDestroy {
    */
   private readonly monacoPromise = firstValueFrom(this.newMonacoEditorCreated.pipe(
```

Comment:
To have a better typing for `monacoPromise`, maybe you can caught error in the observable and use the second parameter of `firstValueFrom` to define `{ defaultValue: undefined }`

Reply:
Currently the typing is `Promise<void | typeof Monaco>`, the `void` comes from the `.catch` so adding `{ defaultValue: undefined }` will change the type to `Promise<void | typeof Monaco | undefined>`
if we want to have `Promise<typeof Monaco | undefined>`, we should either have 
```typescript
(err) => {
      console.error(err);
      return undefined;
}
```
or 
```
.catch<undefined>(...)
```

I checked the implementation of the option `defaultValue` it is used only if the subject `newMonacoEditorCreated` complete before providing a value.
[implementation of `firstValueFrom`](https://github.com/ReactiveX/rxjs/blob/f9f07f14ff15cc5f81e4b34005286efb4b5cabc1/packages/rxjs/src/internal/firstValueFrom.ts#L56)

Reply:
I will go for the second as the issue comes from the `firstValueFrom`.

Reply:
Actually I think defaultValue might better fit my use case here.